src/univariate/continuous/normal.jl

```diff
@@ -1,12 +1,18 @@
-immutable Normal <: ContinuousUnivariateDistribution
-    μ::Float64
-    σ::Float64
+immutable Normal{T<:Number} <: ContinuousUnivariateDistribution
+    μ::T
+    σ::T
 
-    Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); new(μ, σ))
-    Normal(μ::Real) = Normal(μ, 1.0)
-    Normal() = Normal(0.0, 1.0)
+    Normal(μ::T, σ::T) = new(μ, σ)
 end
 
+Normal{T<:Number}(μ::T, σ::T) = Normal{T}(μ, σ)
+Normal(μ::Number, σ::Number) = Normal(promote(μ, σ)...)
+Normal(μ::Number) = Normal(μ, one(μ))
+#Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); Normal(μ, σ))
+Normal(μ::Real) = Normal(μ, one(μ))
+Normal() = Normal(0.0, 1.0)
+Normal(μ::Complex, σ::Complex) = error("Normal with complex mean and variance not defined")
```

Comment:
> To me it feels that calling a GradualNumber to be a Real is a stretch conceptually.

I agree with you that making `GradientNumber`, `HessianNumber`, etc. subtypes of `Real` may not be *algebraically* sound, but it is very *programmatically* sound.

> how about we define the abstract HyperReal type such that Real <: HyperReal <: Number?

While I'm not necessarily against this idea in general, it doesn't solve the problem here. We want functions over `Real`s to accept `ForwardDiffNumber`s. Our options are to:

1. go through everyone's code and submit PRs to change the way their types (we'd still have to do this if we defined `HyperReal`)
2. make our custom type a subtype of  `Real`, which will immediately allow ForwardDiff to be used with a slew of packages that it currently doesn't support

My central argument in favor of option 2 is that it can be done quickly, with no cost to performance or correctness (if we define correctness w.r.t. program behavior). If somebody needs a dual number implementation that resides in the algebraically correct level of the type hierarchy, other packages already exist that provide that.